saleor/core/tasks.py

```diff
@@ -16,18 +17,16 @@ def delete_from_storage_task(path):
 @app.task
 def delete_event_payloads_task():
     delete_period = timezone.now() - settings.EVENT_PAYLOAD_DELETE_PERIOD
-    deliveries = EventDelivery.objects.filter(created_at__lte=delete_period)
-    attempts = EventDeliveryAttempt.objects.filter(
-        Q(Exists(deliveries.filter(id=OuterRef("delivery_id"))))
-        | Q(delivery__isnull=True, created_at__lte=delete_period)
-    )
-    payloads = EventPayload.objects.filter(
-        deliveries__isnull=True, created_at__lte=delete_period
+    deliveries = EventDelivery.objects.filter(created_at__lte=delete_period).order_by(
+        "-pk"
     )
-
-    attempts._raw_delete(attempts.db)
-    deliveries._raw_delete(deliveries.db)
-    payloads._raw_delete(payloads.db)
+    ids = deliveries.values_list("pk", flat=True)[:BATCH_SIZE]
+    qs = EventDelivery.objects.filter(pk__in=ids)
+    if ids:
+        qs.delete()
+        delete_event_payloads_task.delay()
+    else:
+        delete_unused_payloads.delay()
```

Comment:
Do we really need additional task here? 
It looks like it has the same body as `delete_event_payloads_task`, with an additional filter argument. Can't we make a single filter query with `Q()`? 
Also, I think, that there is a mistake in the flow. In the case of calling  `delete_event_payloads_task` -> `delete_unused_payloads`, `delete_unused_payloads` will not re-trigger the task. Which means that we will just delete 100 event payloads no matter how many we have them in DB.